lang/test/org/partiql/lang/eval/ExprValueFactoryTest.kt

```diff
@@ -383,72 +385,54 @@ class ExprValueFactoryTest {
-        val expectedIonValue = createIonTimeStruct(time.hour, time.minute, time.second, time.nano)
-        assertEquals(
-            expected = expectedIonValue,
-            actual = timeIonValue,
-            message = "Expected ionValues to be equal."
-        )
     }
 
     @Test
     fun offsetTimeExprValueTest() {
         val time = LocalTime.of(23, 2, 29, 23)
         val zoneOffset = ZoneOffset.ofTotalSeconds(3600)
         val offsetTime = OffsetTime.of(time, zoneOffset)
-        val timeExprValue = factory.newTime(time, zoneOffset)
+        val timeExprValue = factory.newTime(Time(time, zoneOffset))
         assertEquals(
             expected = offsetTime,
-            actual = timeExprValue.scalar.timeWithTimezoneValue(),
+            actual = timeExprValue.scalar.timeValue()!!.offsetTime,
             message = "Expected values to be equal."
         )
-        val timeIonValue = timeExprValue.ionValue
-        val expectedIonValue = createIonTimeStruct(time.hour, time.minute, time.second, time.nano, zoneOffset.totalSeconds)
-        assertEquals(
-            expected = expectedIonValue,
-            actual = timeIonValue,
-            message = "Expected ionValues to be equal."
-        )
     }
 
     @Test
     fun genericTimeExprValueTest() {
         val timeExprValue = factory.newTime(23, 2, 29, 23, 2)
         assertEquals(
             expected = LocalTime.of(23, 2, 29),
-            actual = timeExprValue.scalar.timeValue(),
+            actual = timeExprValue.scalar.timeValue()!!.localTime,
             message = "Expected values to be equal."
         )
-        val timeIonValue = timeExprValue.ionValue
-        val expectedIonValue = createIonTimeStruct(23, 2, 29)
-        assertEquals(
-            expected = expectedIonValue,
-            actual = timeIonValue,
-            message = "Expected ionValues to be equal."
-        )
     }
 
     @Test
     fun genericTimeExprValueTest2() {
         val timeExprValue = factory.newTime(23, 2, 29, 23, 2, -720)
         assertEquals(
             expected = OffsetTime.of(23, 2, 29, 0, ZoneOffset.ofTotalSeconds(-720*60)),
-            actual = timeExprValue.scalar.timeWithTimezoneValue(),
+            actual = timeExprValue.scalar.timeValue()!!.offsetTime,
             message = "Expected values to be equal."
         )
-        val timeIonValue = timeExprValue.ionValue
-        val expectedIonValue = createIonTimeStruct(23, 2, 29, 0, -720*60)
-        assertEquals(
-            expected = expectedIonValue,
-            actual = timeIonValue,
-            message = "Expected ionValues to be equal."
-        )
+    }
+
+    @Test
+    fun invalidPrecisionForTime() {
+        try {
+            Time.of(23, 12, 34, 344423, -1, 300)
+            Assert.fail("Expected evaluation error")
+        } catch (e: EvaluationException) {
+            Assert.assertEquals(ErrorCode.EVALUATOR_INVALID_PRECISION_FOR_TIME, e.errorCode)
+        }
```

Comment:
Do you have negative tests for a precision of  `>9`?  Do you have negative tests for various invalid times and offsets?  I know you're leaning on the `java.time` library, but we should make sure we have at least some tests here for those cases.

Reply:
That's a great catch. I have added the range checks for all the parameters of `Time.Of` method. I have also added the corresponding negative tests in `test.eval.Time`. I have wrapped around the `DateTimeException` in our own `EvaluationException`.